### PR TITLE
Feature/arbitrary file downloads for oas 3

### PR DESCRIPTION
**Pull request by Geksanit**
part of https://github.com/devexperts/swagger-codegen-ts/issues/73

- embedded files (type: string + format: byte)
- other types such as application/pdf or text/xml
- single file uploads

### CODE REVIEW

**Comment by dxCLA** [comment]
[CLA assistant check](https://cla.devexperts.com/devexperts/swagger-codegen-ts?pullRequest=144) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.devexperts.com/devexperts/swagger-codegen-ts?pullRequest=144) before we can accept your contribution.<br/><hr/>**Alexander Petuhov** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.devexperts.com/check/devexperts/swagger-codegen-ts?pullRequest=144) it.</sub>

**Comment by kokovtsev** [comment]
@Geksanit looks like this change does not 100% cover our need; for example, I tried generating the code using the spec below
```
  /file:
    get:
      tags:
        - files
      operationId: getFile
      responses:
        200:
          description: succesfull operation
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/SomeJsonObject'
            text/csv:
              schema:
                type: string
```

The generated code adds the `Accept: application/json` header to the request and uses the `SomeJsonObject` to decode the response. That is, it is still impossible to download a CSV file.

Let's discuss what interface should be generated for the spec above. As I see it, the consumer should be able to choose the desired response type. Looks like there are two ways:
1. Generate two separate methods:
   ```
	readonly getFileAsApplicationJson: () => Kind2<F, Error, SomeJsonObject>;
	readonly getFileAsTextCsv: () => Kind2<F, Error, string>;
   ```
2. Accept an additional parameter:
   ```
	readonly getFile: (params: {accept: 'application/json'}) => Kind2<F, Error, SomeJsonObject>;
	readonly getFile: (params: {accept: 'text/csv'}) => Kind2<F, Error, string>;
   ```
What do you think?

**Comment by kokovtsev** [comment]
Thanks @Geksanit, looks great except for some minor things! I'm adding some comments in the code

File: src/language/typescript/3.0/serializers/responses-object.ts [hunk 1]
```diff
@@ -35,30 +61,27 @@ export const serializeResponsesObject = (from: Ref) => (
 									() => new Error(`Invalid ${r.$ref} for ResponsesObject'c code "${code}"`),
 								),
 								either.map(getSerializedRefType(from)),
+								either.map(type => [{ mediaType: DEFAULT_MEDIA_TYPE, schema: type }]),
```
[the remark this answers is not in the document]
Comment (Geksanit) [reply src/language/typescript/3.0/serializers/responses-object.ts:64]: Yes, that would be great. But in the current implementation here we have SerializedType (import io and type) and we don't know what media types are inside.

[the remark this answers is not in the document]
Comment (Geksanit) [reply src/language/typescript/3.0/serializers/responses-object.ts:64]: I propose to do it in another task

**Review by kokovtsev — APPROVED** [review]

**Review by kokovtsev — APPROVED** [review]

**Pull request merged**